std/conv.d

```diff
@@ -4121,25 +4121,23 @@ private T octal(T)(const string num)
 {
     assert(isOctalLiteral(num));
 
-    ulong pow = 1;
     T value = 0;
 
-    foreach_reverse (immutable pos; 0 .. num.length)
+    foreach (immutable s; num)
```

Comment:
Use `foreach (const char s; num)` as a note to readers to indicate that the string is not auto-decoded to `dchar`. Also `const` has the same effect as `immutable` for local variables of value types, but it's shorter.